lib/xopts.ts

```diff
@@ -0,0 +1,74 @@
+import {Signer} from 'ethers';
+import {providers} from 'ethers';
+import {ReadOnlyContracts, ReadWriteContracts} from './contracts';
+import {Addresses, Deployments} from './addresses';
+import {ExchangeRate, Currency, MonetaryAmount} from './monetary';
+
+type SignerOrProvider = Signer | providers.Provider;
+
+interface Option<From extends Currency, To extends Currency> {
+  expiry: Date;
+  from: From;
```

Comment:
I would not use the `from` `to`. I would say `underlying`, e.g. BTC and `collateral`, e.g. USDT. Those are the finance terms and IMO less ambiguous.

Reply:
Agreed, I'll update that.